tasks/Base.cpp

```diff
@@ -38,6 +38,15 @@ bool Base::configureHook()
     registerInput("cascade", _timeout_cascade.get(), &_cmd_cascade);
     // Assuming a max number of input ports of 6 (3 for each DOF, linear and angular)
     connected_input_ports.reserve(6);
+    std::vector<auv_control::InputPortConfig> additional_ports = _additional_command_input.get();
+    // Assuming only 4 ports are available since "in" and "cascade" are already registered
+    if (additional_ports.size() > 4){
+        additional_ports.resize(4);
+        std::cerr << "[Warning] Size of the additional input ports is bigger than 4. Taking the first four elements only!" << std::endl;
```

Comment:
From a coding perspective, don't just warn, throw an exception. If the caller is violating a configuration constraint, there will be some serious hair pulling before he/she finds out why the ports don't get created.

But ... why this limit ? In principle, one can create as many as they want. What's unsupported is to use all of them at the same time.

Reply:
The limit is because only 6 degrees of freedom exist. When allowing more than 6, then is has to be monitored when merging the input commands that 6 at most are used.
I could either just remove it, or maybe remove the resize only but keep the warning?

Reply:
You are assuming that all the ports are used all the time. It's easier for Syskit to for instance never use the cascading port and instead use only dynamic ports. Or, to generate all the ports that will be needed and reconnect to the "right" ports as needed, avoiding reconfiguration.

Reply:
> When allowing more than 6, then is has to be monitored when merging the input commands that 6 at most are used.

We actually verify that there are no collisions, that is that there is no two ports setting the same DoF, which is compatible with letting the user create as many ports as he wishes, as long as he/she does not misuse them.

Reply:
alright, removed it.